src/pages/home/report/EmojiPickerMenu.js

```diff
@@ -0,0 +1,122 @@
+import React, {Component} from 'react';
+import {View, FlatList, Text} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import styles from '../../../styles/styles';
+import themeColors from '../../../styles/themes/default';
+import emojis from '../../../../assets/emojis';
+import EmojiPickerMenuItem from './EmojiPickerMenuItem';
+import TextInputFocusable from '../../../components/TextInputFocusable';
+
+const propTypes = {
+    // Controls the visibility of this component.
+    isVisible: PropTypes.bool,
+
+    // Function to add the selected emoji to the main compose text input
+    addEmojiToTextBox: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+    isVisible: false,
+};
+
+class EmojiPickerMenu extends Component {
+    constructor(props) {
+        super(props);
+
+        // Ref for the emoji search input
+        this.searchInput = undefined;
+
+        this.filterEmojis = _.debounce(this.filterEmojis.bind(this), 500, false);
+        this.renderItem = this.renderItem.bind(this);
+
+        this.state = {
+            filteredEmojis: emojis,
+
+            // This is the indices of each category of emojis
+            // The positions are static, and are calculated as index/numColumns (8 in our case)
+            // This is because each row of 8 emojis counts as one index
+            headerIndices: [0, 34, 60, 88, 99, 121, 148],
+        };
+    }
+
+    componentDidUpdate() {
+        if (this.props.isVisible) {
+            this.searchInput.focus();
+        }
+    }
+
+    filterEmojis(searchTerm) {
+        const normalizedSearchTerm = searchTerm.toLowerCase();
+        if (normalizedSearchTerm === '') {
+            this.setState({filteredEmojis: emojis, headerIndices: [0, 34, 60, 88, 99, 121, 148]});
```

Comment:
There are no headers when you search (having categories for individual emojis is weird) so if you don't un-set the headers indices then the first row of emojis will be sticky

Reply:
Makes sense. Might be good to add that as a comment since it might not be obvious that we are "resetting" things back to the initial state here or that removing the `headerIndices` disables to "sticky header" effect.

Reply:
Done :+1: